### PR TITLE
WIP: ossm-3739 Add a maistra-test-tool presubmit integ job

### PR DESCRIPTION
**Pull request by yxun**
This PR includes:
1. A new step registry workflow step of installing jaeger, kiali, servicemesh operators from production OLM on a claimed cluster.
2. A new step registry workflow post step of deleting jaeger, kiali, servicemesh operators when the job is done.
3. Add a maistra-test-tool presubmit job for running `make test` target

### CODE REVIEW

**Comment by yxun** [comment]
> Tests specifics should be performed on the repo, not here.
> 
> Here we should only provision the environment and run the entrypoint of the test (sth like `make test`).

Yes, this new job basically just run `make test` and all details are in the Makefile updated by Marko there.

File: ci-operator/config/maistra/maistra-test-tool/maistra-maistra-test-tool-maistra-2.4.yaml [hunk 1]
```diff
@@ -20,6 +20,51 @@ tests:
...
+      resources:
+        requests:
+          cpu: 100m
+          memory: 100Mi
+      timeout: 20m0s
+    - as: make-test-2-4
+      cli: latest
+      commands: |
+        oc rsh -n "${MAISTRA_NAMESPACE}" "${MAISTRA_SC_POD}" \
+          go install github.com/jstemmer/go-junit-report/v2@latest
```
Comment (jwendell) [inline ci-operator/config/maistra/maistra-test-tool/maistra-maistra-test-tool-maistra-2.4.yaml:55]: The builder image has it.

If that's not enough, tweak the image, or leave this to the test.

File: ci-operator/step-registry/servicemesh/istio/e2e/prod/servicemesh-istio-e2e-prod-commands.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
```
Comment (jwendell) [inline ci-operator/step-registry/servicemesh/istio/e2e/prod/servicemesh-istio-e2e-prod-commands.sh:1]: These should be performed by the test

File: ci-operator/step-registry/servicemesh/istio/post/prod/servicemesh-istio-post-prod-commands.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
```
Comment (jwendell) [inline ci-operator/step-registry/servicemesh/istio/post/prod/servicemesh-istio-post-prod-commands.sh:1]: These should be performed by the test

**Review by jwendell — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/servicemesh/istio/post/prod/servicemesh-istio-post-prod-commands.sh (hunk 3 above)
Comment (yxun) [reply ci-operator/step-registry/servicemesh/istio/post/prod/servicemesh-istio-post-prod-commands.sh:1]: Well, that's a temporary workaround for operators installation from production. 

maistra-test-tool may test pre-release builds (operators from IIB) but we can't leak internal registry source on GH. 
That's why I didn't include any operator installation step in the past and mentioned that operators installation was a Prerequisite before running maistra-test-tool.

Marko may have some idea later and then update the maistra-test-tool for operators step.

File: ci-operator/config/maistra/maistra-test-tool/maistra-maistra-test-tool-maistra-2.4.yaml (hunk 1 above)
Comment (yxun) [reply ci-operator/config/maistra/maistra-test-tool/maistra-maistra-test-tool-maistra-2.4.yaml]: ok, I think it's a GOPATH issue. I will tweak the image later.

File: ci-operator/step-registry/servicemesh/istio/post/prod/servicemesh-istio-post-prod-commands.sh (hunk 3 above)
Comment (yxun) [reply ci-operator/step-registry/servicemesh/istio/post/prod/servicemesh-istio-post-prod-commands.sh:1]: I asked a question about this topic in slack. If they are ok to add a make target, I will remove this "prod-commands" step in this PR later.

Comment (jwendell) [reply ci-operator/step-registry/servicemesh/istio/post/prod/servicemesh-istio-post-prod-commands.sh:1]: What I am trying to say is that we don't need those commands here. Whataver is necessary to to run the tests should be committed in the repo itself, not here. If that's not clear, let's talk on google meet/slack. Ping me.

**Review by jwendell — CHANGES_REQUESTED** [review]

**Comment by yxun** [comment]
I see. I will remove that new workflow step and create a PR in maistra-test-tool repo before testing this PR. Thanks, Jonh.

**Comment by yxun** [comment]
This PR new job depends on https://github.com/maistra/maistra-test-tool/pull/443
When the maistra-test-tool setup PR is merged, we can pj-rehearse test this PR new job.

**Pull request closed without merging**